### PR TITLE
[AIRFLOW-2648] Update mapred job name in HiveOperator

### PR DESCRIPTION
**Pull request by KevinYang21**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-2648) issues and references them in the PR title.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Proposing the change for the following reason:
- hostname within a mapred_job_name does not provide very useful info--dag_id, task_id and execution_date provide good enough observability.
- `Airflow HiveOperator task for` is redundant and makes it hard to parse. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests/operators/hive_operator.py:HiveOperatorTest.test_mapred_job_name

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by KevinYang21** [comment]
Idk what's the deal with `airflow_travis.cfg` but the template won't work inside unless I put `{` instead of `{{`. Suspecting that there's some missing template loading logic. Will open an issue once the problem is identified.

**Comment by Fokko** [comment]
@gglanzani Hive guru, can you take a look?

**Comment by KevinYang21** [comment]
@saguziel PTAL

**Comment by gglanzani** [comment]
@Fokko The hive part looks good to me. 

@yrqls21 I don't understand the reference to `airflow_travis.cfg`. Do you mean that in `default_airflow.cfg` you had to use `{{` while in `airflow_travis.cfg` only `{` would work?

The Jira issue is a bit confusing though. The proposal is to put `host` at the end, but AFAICS that didn't change. Or am I overlooking something?

**Comment by KevinYang21** [comment]
@gglanzani sry I forgot to update the jira. I end up making it configurable so that this won't be a breaking change.

You are right, that's what I mean for the travis.cfg issue.

**Comment by gglanzani** [comment]
@yrqls21 Thanks! That clarifies it.

@Fokko Hive part LGTM, you can decide for the various `.cfg` issues etc.

**Review by Fokko — APPROVED** [review]
LGTM, thanks @yrqls21 @gglanzani

**Comment by KevinYang21** [comment]
I figured out the deal with the CI config, so when we load the default config, [we load the parameterized config which calls `template.format`, and when we load the custom config we read the raw config](https://github.com/apache/incubator-airflow/blob/master/airflow/configuration.py#L448-L450) which doesn't not replace the `{{` with `{`.

I suppose this is expected that we only wanna parameterize default config. @Fokko am I right?
